lib/matplotlib/dates.py

```diff
@@ -444,7 +444,9 @@ def date2num(d):
     if not iterable:
         d = [d]
 
-    d = np.asarray(d)
+    masked = np.any(np.ma.getmask(d))
+    d = np.ma.asarray(d) if masked else np.asarray(d)
```

Comment:
Why not use `np.ma.is_masked(d)`?

Reply:
I have, clearly, limited knowledge of masked arrays... Thanks!